File: apps/app/src/api/charts.ts
import { client } from ".";

export const getScrobblesChart = () => {
  return [];
};

export const getSongChart = async (uri: string) => {
  const response = await client.get(
    "/xrpc/app.rocksky.charts.getScrobblesChart",
    { params: { songuri: uri } },
  );
  if (response.status !== 200) {
    return [];
  }
  return response.data;
};

export const getArtistChart = async (uri: string) => {
  const response = await client.get(
    "/xrpc/app.rocksky.charts.getScrobblesChart",
    { params: { artisturi: uri } },
  );
  if (response.status !== 200) {
    return [];
  }
  return response.data;
};

export const getAlbumChart = async (uri: string) => {
  const response = await client.get(
    "/xrpc/app.rocksky.charts.getScrobblesChart",
    { params: { albumuri: uri } },
  );
  if (response.status !== 200) {
    return [];
  }
  return response.data;
};

export const getProfileChart = async (did: string) => {
  const response = await client.get(
    "/xrpc/app.rocksky.charts.getScrobblesChart",
    { params: { did } },
  );
  if (response.status !== 200) {
    return [];
  }
  return response.data;
};

export const getGenreChart = async (genre: string) => {
  const response = await client.get(
    "/xrpc/app.rocksky.charts.getScrobblesChart",
    { params: { genre } },
  );
  if (response.status !== 200) {
    return [];
  }
  return response.data;
};

File: sdk/elixir/lib/rocksky/charts.ex
defmodule Rocksky.Charts do
  @moduledoc "`app.rocksky.charts.*` endpoints."

  alias Rocksky.HTTP

  @doc """
  Scrobble chart data. Params: `:did`, `:artisturi`, `:albumuri`, `:songuri`,
  `:genre`, `:from`, `:to`.
  """
  def get_scrobbles_chart(client, params \\ []),
    do: HTTP.query(client, "app.rocksky.charts.getScrobblesChart", params)

  @doc "Top artists. Params: `:limit`, `:offset`, `:startDate`, `:endDate`."
  def get_top_artists(client, params \\ []),
    do: HTTP.query(client, "app.rocksky.charts.getTopArtists", params)

  @doc "Top tracks. Params: `:limit`, `:offset`, `:startDate`, `:endDate`."
  def get_top_tracks(client, params \\ []),
    do: HTTP.query(client, "app.rocksky.charts.getTopTracks", params)
end
